common/scylla_configure.py

```diff
@@ -85,10 +84,24 @@ def save_scylla_yaml(self):
     def instance_user_data(self):
         if self._instance_user_data is None:
             try:
-                raw_user_data = self.cloud_instance.user_data
+                raw_user_data = self.cloud_instance.user_data.strip()
                 LOGGER.info("Got user-data: %s", raw_user_data)
-                self._instance_user_data = json.loads(raw_user_data) if raw_user_data.strip() else {}
-                LOGGER.debug("JSON parsed user-data: %s", self._instance_user_data)
+
+                # Try reading mime multipart message, and extract
+                # scylla-machine-image configuration out of it
+                message = message_from_string(raw_user_data)
+                if message.is_multipart():
+                    for i in message.walk():
+                        if i.get_content_type() in ['x-scylla/json', 'x-scylla/yaml']:
+                            raw_user_data = i.get_payload()
+
+                # try parse yaml, and and fallback to parsing json
+                self._instance_user_data = {}
+                if raw_user_data:
+                    self._instance_user_data = yaml.safe_load(raw_user_data)
+                if not self._instance_user_data and raw_user_data:
+                    self._instance_user_data = json.loads(raw_user_data)
+                LOGGER.debug("JSON/YAML parsed user-data: %s", self._instance_user_data)
```

Comment:
I guess would be better to log messages when we know for sure the type.

Reply:
it's not important, at this stage it's already python dict. i'll remove it from the debug print